onnxruntime/python/tools/transformers/models/bert/eval_squad.py

```diff
@@ -190,7 +197,11 @@ def main():
         tokenizer.model_max_length = sequence_length
         tokenizer.doc_stride = min(sequence_length // 2, 128)
 
-        ort_model, onnx_path = load_onnx_model(pretrained_model_name, args.onnx, args.use_gpu)
+        use_gpu = False
+        if args.provider != "CPUExecutionProvider":
```

Comment:
sinces you are not limiting args.provider with [choices](https://docs.python.org/3/library/argparse.html#choices), the `args.provider != "CPUExecutionProvider"` does not implies `ues_gpu=True` here. I think you should change it to `args.provider in ["CUDAExecutionProvider", "ROCMExecutionProvider"]`

Reply:
```py
use_gpu = args.provider in ["CUDAExecutionProvider", "ROCMExecutionProvider"]
```

Reply:
Nevermind, I think you should opt for removing the `--use_gpu` flag from argparse and `use_gpu: bool` in this function all together. Because the newly added provider flag in argparse and provider param in this function serve the same purpose.

Reply:
Done